osu.Game/Database/RealmContextFactory.cs

```diff
@@ -135,23 +135,46 @@ public IDisposable BlockAllOperations()
             if (IsDisposed)
                 throw new ObjectDisposedException(nameof(RealmContextFactory));
 
+            // TODO: this can be added for safety once we figure how to bypass in test
```

Comment:
Possibly. Maybe it doesn't need to even consider the update thread if we rely on this instead. Would you prefer that's fixed in this PR?

Reply:
I guess if someone *does* call this in an improper manner errors will still be thrown, just later on on the realm's side rather than eagerly here. So I'd be fine with leaving until later unless @smoogipoo has a different opinion.